alf/environments/parallel_environment.cpp

```diff
@@ -104,14 +104,17 @@ class SharedDataBuffer {
 };
 
 void CheckStrides(const py::buffer_info& info,
-                  const std::vector<ssize_t>& strides) {
+                  const std::vector<ssize_t>& strides,
+                  const py::buffer& buffer) {
   if (info.strides != strides) {
     throw std::runtime_error(
         py::str(
-            "strides mismatch. Expected: {}, "
+            "Strides mismatch. Expected: {}, "
             "Got: {}, shape: {}. It might be caused by non-continguous array, "
-            "which can be fixed by copying the array.")
-            .format(strides, info.strides, info.shape));
+            "which can be fixed by copying the array. It could also be caused "
+            "by having an array whose dtype is different from its spec. The "
```

Comment:
Instead of suggesting possible dtype mismatch, we can check whether there is in fact dtype mismatch by comparing info.format

Reply:
Good point. Updated. Sorry about having to `rebase` but this PR is small so it should be fine.